src/components/FilterSettings/ColorFilterSettings.svelte

```diff
@@ -34,6 +36,10 @@
     value: field.name,
   }));
 
+  type ColorRuleWithId = ColorRule & { readonly id: string };
+  let conditions: ColorRuleWithId[];
+  $: conditions = filter.conditions.map((cond) => ({ ...cond, id: uuidv4() }));
```

Comment:
Add a new type `ColorRuleWithId` for both iteration and dnd, ids are assigned using `uuidv4()` **everytime** filter updates